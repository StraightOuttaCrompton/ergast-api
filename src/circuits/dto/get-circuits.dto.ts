import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";

export class GetCircuitsDto {
    /**
     * TODO: extract into class which this inherits from
     */
    @IsOptional()
    @IsInt()
    @IsPositive()
    offset: number;

    @IsOptional()
    @IsInt()
    @IsPositive()
    limit: number;
    /**
     * --------------------------------------------------
     */

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
