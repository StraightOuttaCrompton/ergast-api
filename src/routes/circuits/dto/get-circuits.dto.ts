import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetCircuitsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
