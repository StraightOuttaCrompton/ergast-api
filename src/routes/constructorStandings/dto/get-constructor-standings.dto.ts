import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
