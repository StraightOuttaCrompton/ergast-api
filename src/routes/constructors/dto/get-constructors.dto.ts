import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number; // TODO: validate for actual status values

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    constructorStandings?: number;
}
