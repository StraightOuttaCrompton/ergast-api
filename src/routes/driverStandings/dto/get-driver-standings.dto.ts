import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriverStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
