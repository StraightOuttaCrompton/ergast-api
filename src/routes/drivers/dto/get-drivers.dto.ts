import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriversDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    driverStandings?: number;
}
