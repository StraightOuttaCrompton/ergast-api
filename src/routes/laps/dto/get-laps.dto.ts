import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetLapsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;
}
