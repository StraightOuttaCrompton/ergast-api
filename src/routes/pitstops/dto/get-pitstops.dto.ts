import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetPitstopsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    lapNumber: number;
}
