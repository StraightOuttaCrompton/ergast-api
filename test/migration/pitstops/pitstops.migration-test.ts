import querystring from "querystring";
import { getMigrationTest } from "../migrationUtils";
import { GetPitstopsDto } from "src/routes/pitstops/dto/get-pitstops.dto";
import path from "path";

describe("GET /pitstops/${pitNumber} smoke tests", () => {
    const migrationTest = getMigrationTest(__dirname, path.basename(__filename), (response) => {
        const { season, round, ...rest } = response.MRData.RaceTable.Races[0];

        return rest;
    });

    migrationTest(
        `/pitstops/1?${querystring.stringify({ year: 2022, round: 8 })}`,
        `https://ergast.com/api/f1/2022/8/pitstops/1.json`
    );
});

describe("GET /pitstops smoke tests", () => {
    const migrationTest = getMigrationTest(__dirname, path.basename(__filename), (response) => {
        const { season, round, ...rest } = response.MRData.RaceTable.Races[0];

        return rest;
    });
    const endpoint = "/pitstops";

    const queries: { queryParams: Partial<GetPitstopsDto>; legacyRoute: string }[] = [
        {
            queryParams: { year: 2022, round: 3 },
            legacyRoute: "2022/3",
        },
        {
            queryParams: { year: 2011, round: 5, driverId: "alonso" },
            legacyRoute: "2011/5/drivers/alonso",
        },
        {
            queryParams: { year: 2020, round: 8, lapNumber: 20 },
            legacyRoute: "2020/8/laps/20",
        },
    ];

    queries.forEach(({ queryParams, legacyRoute }) => {
        migrationTest(
            `${endpoint}?${querystring.stringify(queryParams)}`,
            `https://ergast.com/api/f1/${legacyRoute}/pitstops.json`
        );
    });
});
