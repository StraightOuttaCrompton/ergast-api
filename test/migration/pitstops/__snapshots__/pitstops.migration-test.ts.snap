// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /pitstops smoke tests /pitstops?year=2011&round=5&driverId=alonso 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Spain",
      "lat": "41.57",
      "locality": "Montmel√≥",
      "long": "2.26111",
    },
    "circuitId": "catalunya",
    "circuitName": "Circuit de Barcelona-Catalunya",
    "url": "http://en.wikipedia.org/wiki/Circuit_de_Barcelona-Catalunya",
  },
  "PitStops": [
    {
      "driverId": "alonso",
      "duration": "20.567",
      "lap": "10",
      "stop": "1",
      "time": "14:18:23",
    },
    {
      "driverId": "alonso",
      "duration": "20.238",
      "lap": "19",
      "stop": "2",
      "time": "14:32:13",
    },
    {
      "driverId": "alonso",
      "duration": "20.738",
      "lap": "29",
      "stop": "3",
      "time": "14:47:26",
    },
    {
      "driverId": "alonso",
      "duration": "20.380",
      "lap": "39",
      "stop": "4",
      "time": "15:02:54",
    },
  ],
  "date": "2011-05-22",
  "raceName": "Spanish Grand Prix",
  "time": "12:00:00Z",
  "url": "http://en.wikipedia.org/wiki/2011_Spanish_Grand_Prix",
}
`;

exports[`GET /pitstops smoke tests /pitstops?year=2020&round=8&lapNumber=20 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Italy",
      "lat": "45.6156",
      "locality": "Monza",
      "long": "9.28111",
    },
    "circuitId": "monza",
    "circuitName": "Autodromo Nazionale di Monza",
    "url": "http://en.wikipedia.org/wiki/Autodromo_Nazionale_Monza",
  },
  "PitStops": [
    {
      "driverId": "hamilton",
      "duration": "24.671",
      "lap": "20",
      "stop": "1",
      "time": "15:41:58",
    },
    {
      "driverId": "giovinazzi",
      "duration": "23.998",
      "lap": "20",
      "stop": "1",
      "time": "15:42:53",
    },
  ],
  "date": "2020-09-06",
  "raceName": "Italian Grand Prix",
  "time": "13:10:00Z",
  "url": "http://en.wikipedia.org/wiki/2020_Italian_Grand_Prix",
}
`;

exports[`GET /pitstops smoke tests /pitstops?year=2022&round=3 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Australia",
      "lat": "-37.8497",
      "locality": "Melbourne",
      "long": "144.968",
    },
    "circuitId": "albert_park",
    "circuitName": "Albert Park Grand Prix Circuit",
    "url": "http://en.wikipedia.org/wiki/Melbourne_Grand_Prix_Circuit",
  },
  "PitStops": [
    {
      "driverId": "stroll",
      "duration": "18.242",
      "lap": "3",
      "stop": "1",
      "time": "15:08:15",
    },
    {
      "driverId": "stroll",
      "duration": "18.560",
      "lap": "4",
      "stop": "2",
      "time": "15:10:29",
    },
    {
      "driverId": "latifi",
      "duration": "18.552",
      "lap": "13",
      "stop": "1",
      "time": "15:25:03",
    },
    {
      "driverId": "mick_schumacher",
      "duration": "19.039",
      "lap": "14",
      "stop": "1",
      "time": "15:26:23",
    },
    {
      "driverId": "ocon",
      "duration": "19.128",
      "lap": "17",
      "stop": "1",
      "time": "15:30:30",
    },
    {
      "driverId": "max_verstappen",
      "duration": "18.281",
      "lap": "18",
      "stop": "1",
      "time": "15:31:37",
    },
    {
      "driverId": "tsunoda",
      "duration": "18.443",
      "lap": "18",
      "stop": "1",
      "time": "15:32:02",
    },
    {
      "driverId": "perez",
      "duration": "17.434",
      "lap": "20",
      "stop": "1",
      "time": "15:34:33",
    },
    {
      "driverId": "norris",
      "duration": "17.556",
      "lap": "20",
      "stop": "1",
      "time": "15:34:40",
    },
    {
      "driverId": "ricciardo",
      "duration": "18.150",
      "lap": "21",
      "stop": "1",
      "time": "15:36:06",
    },
    {
      "driverId": "gasly",
      "duration": "18.926",
      "lap": "21",
      "stop": "1",
      "time": "15:36:15",
    },
    {
      "driverId": "zhou",
      "duration": "18.380",
      "lap": "21",
      "stop": "1",
      "time": "15:36:22",
    },
    {
      "driverId": "leclerc",
      "duration": "17.973",
      "lap": "22",
      "stop": "1",
      "time": "15:37:04",
    },
    {
      "driverId": "hamilton",
      "duration": "17.877",
      "lap": "22",
      "stop": "1",
      "time": "15:37:22",
    },
    {
      "driverId": "bottas",
      "duration": "18.413",
      "lap": "22",
      "stop": "1",
      "time": "15:37:40",
    },
    {
      "driverId": "russell",
      "duration": "18.479",
      "lap": "23",
      "stop": "1",
      "time": "15:38:51",
    },
    {
      "driverId": "stroll",
      "duration": "20.402",
      "lap": "23",
      "stop": "3",
      "time": "15:39:30",
    },
    {
      "driverId": "latifi",
      "duration": "17.936",
      "lap": "23",
      "stop": "2",
      "time": "15:40:00",
    },
    {
      "driverId": "alonso",
      "duration": "18.796",
      "lap": "39",
      "stop": "1",
      "time": "16:05:24",
    },
    {
      "driverId": "kevin_magnussen",
      "duration": "19.111",
      "lap": "39",
      "stop": "1",
      "time": "16:05:30",
    },
    {
      "driverId": "alonso",
      "duration": "18.394",
      "lap": "53",
      "stop": "2",
      "time": "16:25:28",
    },
    {
      "driverId": "albon",
      "duration": "19.845",
      "lap": "57",
      "stop": "1",
      "time": "16:30:33",
    },
  ],
  "date": "2022-04-10",
  "raceName": "Australian Grand Prix",
  "time": "05:00:00Z",
  "url": "http://en.wikipedia.org/wiki/2022_Australian_Grand_Prix",
}
`;

exports[`GET /pitstops/\${pitNumber} smoke tests /pitstops/1?year=2022&round=8 1`] = `
{
  "Circuit": {
    "Location": {
      "country": "Azerbaijan",
      "lat": "40.3725",
      "locality": "Baku",
      "long": "49.8533",
    },
    "circuitId": "baku",
    "circuitName": "Baku City Circuit",
    "url": "http://en.wikipedia.org/wiki/Baku_City_Circuit",
  },
  "PitStops": [
    {
      "driverId": "latifi",
      "duration": "29.395",
      "lap": "3",
      "stop": "1",
      "time": "15:09:50",
    },
    {
      "driverId": "leclerc",
      "duration": "23.212",
      "lap": "9",
      "stop": "1",
      "time": "15:20:26",
    },
    {
      "driverId": "russell",
      "duration": "20.538",
      "lap": "9",
      "stop": "1",
      "time": "15:20:43",
    },
    {
      "driverId": "gasly",
      "duration": "22.698",
      "lap": "9",
      "stop": "1",
      "time": "15:20:49",
    },
    {
      "driverId": "hamilton",
      "duration": "21.470",
      "lap": "9",
      "stop": "1",
      "time": "15:20:54",
    },
    {
      "driverId": "vettel",
      "duration": "20.249",
      "lap": "9",
      "stop": "1",
      "time": "15:20:55",
    },
    {
      "driverId": "tsunoda",
      "duration": "23.083",
      "lap": "9",
      "stop": "1",
      "time": "15:20:57",
    },
    {
      "driverId": "zhou",
      "duration": "23.001",
      "lap": "9",
      "stop": "1",
      "time": "15:21:08",
    },
    {
      "driverId": "kevin_magnussen",
      "duration": "21.555",
      "lap": "9",
      "stop": "1",
      "time": "15:21:11",
    },
    {
      "driverId": "albon",
      "duration": "25.577",
      "lap": "9",
      "stop": "1",
      "time": "15:21:16",
    },
    {
      "driverId": "mick_schumacher",
      "duration": "26.973",
      "lap": "9",
      "stop": "1",
      "time": "15:21:23",
    },
    {
      "driverId": "perez",
      "duration": "23.379",
      "lap": "16",
      "stop": "1",
      "time": "15:33:43",
    },
    {
      "driverId": "max_verstappen",
      "duration": "21.468",
      "lap": "18",
      "stop": "1",
      "time": "15:37:17",
    },
    {
      "driverId": "alonso",
      "duration": "20.571",
      "lap": "18",
      "stop": "1",
      "time": "15:37:59",
    },
    {
      "driverId": "norris",
      "duration": "21.443",
      "lap": "20",
      "stop": "1",
      "time": "15:41:42",
    },
    {
      "driverId": "stroll",
      "duration": "22.059",
      "lap": "30",
      "stop": "1",
      "time": "16:00:30",
    },
    {
      "driverId": "ricciardo",
      "duration": "21.794",
      "lap": "33",
      "stop": "1",
      "time": "16:06:02",
    },
    {
      "driverId": "ocon",
      "duration": "22.638",
      "lap": "33",
      "stop": "1",
      "time": "16:06:22",
    },
    {
      "driverId": "bottas",
      "duration": "20.979",
      "lap": "33",
      "stop": "1",
      "time": "16:06:34",
    },
  ],
  "date": "2022-06-12",
  "raceName": "Azerbaijan Grand Prix",
  "time": "11:00:00Z",
  "url": "http://en.wikipedia.org/wiki/2022_Azerbaijan_Grand_Prix",
}
`;
